app/api/(auth)/register/route.ts

```diff
@@ -0,0 +1,56 @@
+import bcrypt from 'bcrypt';
+import { redirect } from 'next/navigation';
+import { NextRequest, NextResponse } from 'next/server';
+import { z } from 'zod';
+import { createUser, getUserByUsername } from '../../../../database/users';
+
+const userType = z.object({
+  username: z.string(),
+  password: z.string(),
+});
+
+export type LoginResponseBody =
+  | { errors: { message: string }[] }
+  | { user: { username: string } };
+
+export const POST = async (request: NextRequest) => {
+  const body = await request.json();
+
+  const result = userType.safeParse(body);
+
+  if (!result.success) {
+    // Inside of result.error.issues you are going to have more granular information about what is failing allowing you to create more specific error massages
+    // console.log(result.error.issues);
+
+    return NextResponse.json(
+      {
+        error:
```

Comment:
- types above say response key should be `errors`, not `error
- you could return the error.issues here from Zod

Reply:
b92e630

Reply:
5bea583